bloom-housing/bloom
# feat: correction script for sparks homes application preferences

This PR addresses #(794)


## Description

Script collects the existing applications to Sparks Homes and updates the preferences jsonb for each application with corrected options strings.

## How Can This Be Tested/Reviewed?

Pull down prob db copy from Heroku and setup locally.
Check the preferences field for applications associated with listing id 'a055ce66-a074-4f3a-b67b-6776bec9926e'
Start app, sign in as an admin through partner portal or api.
Through the api, call `/scriptRunner/correctApplicationPreferenceDataForSparksHomes`.
Check the preferences field for applications associated with listing id 'a055ce66-a074-4f3a-b67b-6776bec9926e'
and assert update worked.

query:
`SELECT preferences
FROM applications
WHERE listing_id='a055ce66-a074-4f3a-b67b-6776bec9926e';`

## Author Checklist:


## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
